app/models/processor.rb

```diff
@@ -0,0 +1,80 @@
+require 'json'
+require 'uri'
+require 'pry'
+
+class Processor < ActiveRecord::Base
```

Comment:
we already talked a little bit about how this class has a lot of repetitive code. see if you can extract this out into smaller methods so you can reuse this logic!